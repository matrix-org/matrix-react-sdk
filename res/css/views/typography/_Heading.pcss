/*
Copyright 2021 - 2023x The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.mx_Heading_h1,
.mx_Heading_h2,
.mx_Heading_h3,
.mx_Heading_h4 {
    margin-inline: unset;
    margin-block: unset;
}

.mx_Heading_h1 {
    font: var(--cpd-font-heading-xl-semibold);
}

.mx_Heading_h2 {
    font: var(--cpd-font-heading-lg-semibold);
}

.mx_Heading_h3 {
    font: var(--cpd-font-heading-md-semibold);
    font-weight: var(--cpd-font-weight-semibold);
}

.mx_Heading_h4 {
    font: var(--cpd-font-heading-sm-semibold);
    font-weight: var(--cpd-font-weight-semibold);
}
