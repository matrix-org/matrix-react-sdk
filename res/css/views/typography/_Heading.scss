.mx_Heading_h1 {
    font-size: $font-32px;
    font-weight: $font-semi-bold;
    line-height: $font-39px;
    margin-inline: unset;
    margin-block: unset;
}

.mx_Heading_h2 {
    font-size: $font-24px;
    font-weight: $font-semi-bold;
    line-height: $font-29px;
    margin-inline: unset;
    margin-block: unset;
}

.mx_Heading_h3 {
    font-size: $font-18px;
    font-weight: $font-semi-bold;
    line-height: $font-22px;
    margin-inline: unset;
    margin-block: unset;
}
