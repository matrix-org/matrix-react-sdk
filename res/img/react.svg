<svg xmlns="http://www.w3.org/2000/svg" width="18" height="17" viewBox="0 0 18 17">
    <g fill="none" fill-rule="evenodd">
        <path fill="#2E2F32" fill-rule="nonzero" d="M13.6 14.5a7.875 7.875 0 1 1 2.183-3h-.818a7.125 7.125 0 1 0-2.62 3H13.6z"/>
        <g stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width=".75">
            <path d="M11.5 13h6M14.5 10v6"/>
        </g>
        <path stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width=".75" d="M5.5 10s1.125 1.5 3 1.5 3-1.5 3-1.5"/>
        <path stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M6.25 6.25h.008M10.75 6.25h.008"/>
    </g>
</svg>
