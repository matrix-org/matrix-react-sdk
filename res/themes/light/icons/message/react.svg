<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 464c-119.1 0-216-96.9-216-216S128.9 40 248 40s216 96.9 216 216-96.9 216-216 216zm90.2-146.2C315.8 352.6 282.9 368 248 368s-67.8-15.4-90.2-42.2c-5.7-6.8-15.8-7.7-22.5-2-6.8 5.7-7.7 15.7-2 22.5C161.7 380.4 203.6 400 248 400s86.3-19.6 114.8-53.8c5.7-6.8 4.8-16.9-2-22.5-6.8-5.6-16.9-4.7-22.6 2.1zM168 240c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160 0c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32z"/></svg>