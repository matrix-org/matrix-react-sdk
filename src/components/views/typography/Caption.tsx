/*
Copyright 2022 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classNames from "classnames";
import React, { HTMLAttributes } from "react";

interface Props extends Omit<HTMLAttributes<HTMLSpanElement>, "className"> {
    children: React.ReactNode;
    isError?: boolean;
}

export const Caption: React.FC<Props> = ({ children, isError, ...rest }) => {
    return (
        <span
            className={classNames("mx_Caption", {
                mx_Caption_error: isError,
            })}
            {...rest}
        >
            {children}
        </span>
    );
};
