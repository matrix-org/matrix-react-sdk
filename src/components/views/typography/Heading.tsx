/*
Copyright 2021 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { HTMLAttributes } from "react";
import classNames from "classnames";

type Size = "1" | "2" | "3" | "4";

type HTMLHeadingTags = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
    /**
     * Defines the type of heading used
     */
    as?: HTMLHeadingTags;
    /**
     * Defines the appearance of the heading
     * Falls back to the type of heading used if `as` is not provided
     */
    size: Size;
}

const Heading: React.FC<HeadingProps> = ({ as, size = "1", className, children, ...rest }) =>
    React.createElement(as || `h${size}`, {
        ...rest,
        className: classNames(`mx_Heading_h${size}`, className),
        children,
    });

export default Heading;
