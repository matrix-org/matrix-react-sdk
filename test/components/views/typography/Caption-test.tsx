/*
Copyright 2022 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { render } from "@testing-library/react";

import { Caption } from "../../../../src/components/views/typography/Caption";

describe("<Caption />", () => {
    const defaultProps = {
        "children": "test",
        "data-testid": "test test id",
    };
    const getComponent = (props = {}) => <Caption {...defaultProps} {...props} />;

    it("renders plain text children", () => {
        const { container } = render(getComponent());
        expect({ container }).toMatchSnapshot();
    });

    it("renders an error message", () => {
        const { container } = render(getComponent({ isError: true }));
        expect({ container }).toMatchSnapshot();
    });

    it("renders react children", () => {
        const children = (
            <>
                Test <b>test but bold</b>
            </>
        );
        const { container } = render(getComponent({ children }));
        expect({ container }).toMatchSnapshot();
    });
});
