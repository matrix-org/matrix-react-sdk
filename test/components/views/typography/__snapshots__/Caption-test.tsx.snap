// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption /> renders plain text children 1`] = `
{
  "container": <div>
    <span
      class="mx_Caption"
      data-testid="test test id"
    >
      test
    </span>
  </div>,
}
`;

exports[`<Caption /> renders react children 1`] = `
{
  "container": <div>
    <span
      class="mx_Caption"
      data-testid="test test id"
    >
      Test 
      <b>
        test but bold
      </b>
    </span>
  </div>,
}
`;
